chart-service/database/host_cores.go

```diff
@@ -0,0 +1,51 @@
+package database
+
+import (
+	"context"
+	"github.com/amreo/mu"
+	"github.com/ercole-io/ercole/v2/chart-service/dto"
+	"github.com/ercole-io/ercole/v2/utils"
+	"go.mongodb.org/mongo-driver/bson"
+	"time"
+)
+
+func (md *MongoDatabase) GetHostCores(location string, environment string, olderThan time.Time, newerThan time.Time) ([]dto.HostCores, error) {
+	var items []dto.HostCores
```

Comment:
Why declaring it here when is only needed at line 47?

Reply:
If you declare the slice of output without initialize it, if there's no result it will return nil instead of an empty slice, which is instead preferable: in fact this wouldn't be an error but an empty result!